models/login_source.go

```diff
@@ -576,17 +629,17 @@ func UserSignIn(username, password string) (*User, error) {
 				return nil, ErrLoginSourceNotExist{user.LoginSource}
 			}
 
-			return ExternalUserLogin(user, user.LoginName, password, &source, false)
+			return ExternalUserLogin(user, user.LoginName, password, &source, false, ctx, sess)
 		}
 	}
 
-	sources := make([]*LoginSource, 0, 3)
+	sources := make([]*LoginSource, 0, 5)
```

Comment:
You added a single LoginSource but you incremented this number by 2, either there's a bug in current code (short count) or in your patch (long count) ?